src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -177,6 +185,52 @@ public PlayerMock(ServerMock server, String name, UUID uuid)
 		closeInventory();
 	}
 
+	/**
+	 * Simulates a disconnection from the server.
+	 *
+	 * @return True if the player was disconnected, false if they were already offline.
+	 */
+	public boolean disconnect()
+	{
+		if (!online)
+		{
+			return false;
+		}
+		this.online = false;
+		this.lastPlayed = System.currentTimeMillis();
+
+		PlayerQuitEvent playerQuitEvent = new PlayerQuitEvent(this, String.format(QUIT_MESSAGE, this.getDisplayName()));
+		Bukkit.getPluginManager().callEvent(playerQuitEvent);
+
+		this.server.getPlayerList().disconnectPlayer(this);
+
+		return true;
+	}
+
+	/**
+	 * Simulates a connection to the server.
+	 *
+	 * @return True if the player was connected, false if they were already online.
+	 */
+	public boolean reconnect()
+	{
+		if (firstPlayed == 0)
+		{
+			throw new IllegalStateException("Player wasn't online before");
```

Comment:
```suggestion
			throw new IllegalStateException("Player isn't online");
```

Reply:
That would be wrong in this case, since firstPlayer = 0 is only true if the Player hasn't played before

Reply:
I see, reworded it